Week 5/Solution2.java

```diff
@@ -0,0 +1,109 @@
+import java.io.*;
+import java.util.*;
+import org.junit.Test;
+import static org.junit.Assert.assertEquals;
+//This class is the implementation of the alternative method with O(1) space usage.
+//Implies the modification of the list as suggested by Connie and has the fast pointer method to find the middle node.
+public class Solution2 {
+	//Node structure considered for this problem
+	class Node{
+		char value;
+		Node next;
+		Node(char value, Node next){
+			this.value = value;
+			this.next = next;
+		}
+	}
```

Comment:
nit: remember to leave an extra newline between different classes/methods etc.

Reply:
Will do, thanks!